crates/server/src/admin/utils/jwt.rs

```diff
@@ -0,0 +1,230 @@
+    .map_err(|err| ApiError {
+        status_code: StatusCode::BAD_REQUEST,
+        message: format!("Failed to generate access token: {}", err),
+    })?;
+
+    // Generate Refresh Token
+    let refresh_expiration = Utc::now() + Duration::days(30);
+    let refresh_claims = Claims {
+        context_id: context_id.to_string(),
+        exec_pub_key: executor_public_key.to_string(),
+        exp: refresh_expiration.timestamp() as usize,
+        token_type: "refresh".to_string(),
+    };
+
+    let refresh_token = encode(
+        &Header::default(),
+        &refresh_claims,
+        &EncodingKey::from_secret(jwt_secret.as_slice()),
+    )
+    .map_err(|err| ApiError {
+        status_code: StatusCode::BAD_REQUEST,
+        message: format!("Failed to generate refresh token: {}", err),
+    })?;
+
+    // Store the refresh token in the database
+    create_or_update_refresh_token(store.clone(), refresh_token.as_bytes().to_vec()).map_err(
+        |err| ApiError {
+            status_code: StatusCode::INTERNAL_SERVER_ERROR,
+            message: format!("Failed to store refresh token: {}", err),
+        },
+    )?;
+
+    Ok(JwtToken {
+        access_token,
+        refresh_token,
+    })
+}
+
+// Check if the refresh token is valid and generate new tokens
+pub fn refresh_access_token(refresh_token: &str, store: Store) -> Result<JwtToken, ApiError> {
+
+    // Get the JWT secret from the DB
+    let jwt_secret = match get_jwt_secret(store.clone()) {
+        Ok(Some(secret)) => secret.jwt_secret().to_vec(),
+        Ok(None) => {
+            return Err(ApiError {
+                status_code: StatusCode::INTERNAL_SERVER_ERROR,
+                message: "JWT secret not found".into(),
+            });
+        }
+        Err(err) => {
+            return Err(ApiError {
+                status_code: StatusCode::INTERNAL_SERVER_ERROR,
+                message: format!("Failed to get JWT secret: {}", err),
+            });
+        }
+    };
+    // Check if the refresh token from the database is present
+    let refresh_token_db = match get_refresh_token(store.clone()) {
+        Ok(Some(token)) => {
+            let refresh_token = from_utf8(token.refresh_token()).map_err(|err| ApiError {
+                status_code: StatusCode::INTERNAL_SERVER_ERROR,
+                message: format!("Failed to parse refresh token: {}", err),
+            })?;
+            refresh_token.to_string()
+        }
+        Ok(None) => {
+            return Err(ApiError {
+                status_code: StatusCode::FORBIDDEN,
+                message: "Refresh token not found".into(),
+            });
+        }
+        Err(err) => {
+            return Err(ApiError {
+                status_code: StatusCode::INTERNAL_SERVER_ERROR,
+                message: format!("Failed to get refresh token: {}", err),
+            });
+        }
+    };
+    // Check if the refresh token from the store matches the provided one
+    if refresh_token_db != refresh_token {
```

Comment:
in the case we have multiple clients, we'd want to be able to refresh multiple access tokens, so we can't store just one refresh token